runtime/src/accounts.rs

```diff
@@ -1367,38 +1367,6 @@ mod tests {
         }
     }
 
-    #[test]
-    fn test_load_account_pay_to_self() {
-        let mut accounts: Vec<(Pubkey, Account)> = Vec::new();
-        let mut error_counters = ErrorCounters::default();
-
-        let keypair = Keypair::new();
-        let pubkey = keypair.pubkey();
-
-        let account = Account::new(10, 1, &Pubkey::default());
-        accounts.push((pubkey, account));
-
-        let instructions = vec![CompiledInstruction::new(0, &(), vec![0, 1])];
-        // Simulate pay-to-self transaction, which loads the same account twice
-        let tx = Transaction::new_with_compiled_instructions(
-            &[&keypair],
-            &[pubkey],
-            Hash::default(),
-            vec![native_loader::id()],
-            instructions,
-        );
-        let loaded_accounts = load_accounts(tx, &accounts, &mut error_counters);
-
-        assert_eq!(loaded_accounts.len(), 1);
-        assert_eq!(
-            loaded_accounts[0],
-            (
-                Err(TransactionError::InvalidAccountForFee),
```

Comment:
We see here f37f83fd12 that after we allowed duplicate accounts, we saw the original duplicate account error was no longer valid, but instead of removing the test, the error was updated to a different error triggered by this test code.  That's clearly not the intent of this test, so deleting it.